zapisy/apps/theses/views.py

```diff
@@ -59,21 +59,16 @@ def list_all(request):
 @login_required
 def view_thesis(request, id):
     """Show subpage for one thesis."""
-    thesis = get_object_or_404(Thesis, id=id)
+    thesis = get_object_or_404(Thesis.objects.visible(request.user), id=id)
     board_member = is_theses_board_member(request.user)
 
-    user_privileged_for_thesis = thesis.is_among_advisors(
-        request.user) or request.user.is_staff or board_member
-
-    if not thesis.has_been_accepted and not user_privileged_for_thesis:
-        raise PermissionDenied
+    user_privileged_for_thesis = thesis.is_user_privileged(request.user)
     can_edit_thesis = thesis.is_mine(request.user)
     save_and_verify = thesis.is_mine(request.user) and thesis.is_returned
     can_vote = thesis.is_voting_active and board_member
```

Comment:
Użyta zmienna zrobiła się już trochę niepotrzebna i można się pozbyć jej deklaracji.
```suggestion
    can_vote = thesis.is_voting_active and is_theses_board_member(request.user)
```

Reply:
Zmienna `board_member` jest nadal używana w warunku `if` ok. 40 linii niżej